arbor/simulation.cpp

```diff
@@ -558,10 +565,41 @@ void simulation::set_local_spike_callback(spike_export_function export_callback)
     impl_->local_export_callback_ = std::move(export_callback);
 }
 
+void simulation::set_epoch_callback(epoch_function epoch_callback, bool global) {
+    impl_->epoch_callback_ = std::move(epoch_callback);
+}
+
 void simulation::inject_events(const cse_vector& events) {
     impl_->inject_events(events);
 }
 
 simulation::~simulation() = default;
 
+epoch_function epoch_progress_bar() {
+    struct impl {
+        double t0 = 0;
+        unsigned calls = 0;
+
+        const char* PBSTR = "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||";
+        unsigned PBWIDTH = 50;
```

Comment:
```suggestion
        unsigned PBWIDTH = 50;
        const std::string PBSTR = std::string(2*PBWIDTH, '|');
```
Using the "repeat character" constructor of `std::string`.

Reply:
Seems like the best solution!